### PR TITLE
S3262 FN: Mark comments as compliant in UTs 

### LINKED ISSUE #6389: S3262 FN: Mark comments as compliant in UTs
### Description

FN for S3262 (MethodOverrideNoParams).

Using C#11 static virtual/abstract methods in interfaces, the rule is not raising an issue.

### Repro steps

```csharp
            public interface IMath
{
    static virtual void Method(params int[] numbers) { }
    static virtual void Method(string s, params int[] numbers) { }
    static abstract void Method(string s, string s1, params int[] numbers);
}

public class Foo : IMath
{
    public static void Method(int[] numbers) { } // Compliant, FN
    public static void Method(string s, int[] numbers) { } // Compliant, FN
    public static void Method(string s, string s1, int[] numbers) { } // Compliant, FN
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

### PR DESCRIPTION
**Pull request by gregory-paidis-sonarsource**
Fixes #6389

### CODE REVIEW

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverrideNoParams.cs [hunk 1]
```diff
@@ -26,32 +26,31 @@ public sealed class MethodOverrideNoParams : SonarDiagnosticAnalyzer
...
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
                     var method = (MethodDeclarationSyntax)c.Node;
                     var methodSymbol = c.SemanticModel.GetDeclaredSymbol(method);
-                    if (methodSymbol == null ||
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverrideNoParams.cs:41]: Actually if you invert the conditions you can merge it with the if below which I think should increase the readability. WDYT?

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
Left one Nitpick comment

File: analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverrideNoParams.cs [hunk 2]
```diff
@@ -26,32 +26,27 @@ public sealed class MethodOverrideNoParams : SonarDiagnosticAnalyzer
...
-                    {
-                        return;
-                    }
 
-                    if (!method.ParameterList.Parameters.Last().Modifiers.Any(SyntaxKind.ParamsKeyword))
+                    if (methodSymbol is not null
+                    && methodSymbol.IsOverride
+                    && methodSymbol.OverriddenMethod is not null
+                    && methodSymbol.OverriddenMethod.Parameters.Any(p => p.IsParams)
+                    && !method.ParameterList.Parameters.Last().Modifiers.Any(SyntaxKind.ParamsKeyword))
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverrideNoParams.cs:45]: It should be indented with one more tab.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
I approved with one indentation to fix. Please address it before merging.

**Pull request merged**